volumes.cpp

```diff
@@ -1312,7 +1331,249 @@ SurfaceVolume& makeSurface( const SurfaceCard* card, VolumeCache* v){
     
   }
   
-  
 }
 
+SurfaceVolume* FacetSurface( const std::string mnemonic, const std::vector< double > args, int facet ){
+  if( mnemonic == "rcc" ){
+    return rccFacet( args, facet );
+  }
+  else if(mnemonic == "box"){
+    return boxFacet( args, facet );
+  }
+  else if( mnemonic == "rpp"){
+    return rppFacet( args, facet );
+  }
+  else if( mnemonic == "hex" || mnemonic == "rhp" ){
+    return hexFacet( args, facet );
+  }
+  else if( mnemonic == "rec" ){
+    return recFacet( args, facet );
+  }
+  throw std::runtime_error( mnemonic + " does not have macrobody facet support at this time." );
+}    
+
+
+//This function is used to find D in the planar equation Ax + By + Cz - D = 0 when given two vectors.
+int planePoint( std::vector< double > args, int i, bool end ){
```

Comment:
One vector gives a starting position.  The plane is then made perpendicular to the other vector, at the end of the second vector (which begins at the end of the first vector) if end is true, otherwise at the starting position.